compiler/rustc_lint/src/builtin.rs

```diff
@@ -2223,6 +2241,8 @@ impl<'tcx> LateLintPass<'tcx> for ExplicitOutlivesRequirements {
                     Applicability::MaybeIncorrect
                 };
 
+                let lint_spans = lint_spans.into_iter().collect::<Vec<_>>();
+
                 cx.emit_spanned_lint(
                     EXPLICIT_OUTLIVES_REQUIREMENTS,
                     lint_spans.clone(),
```

Comment:
Remove `collect`? also the clone seems unnecessary? Can't this just be 
```suggestion
                cx.emit_spanned_lint(
                    EXPLICIT_OUTLIVES_REQUIREMENTS,
                    lint_spans,
```

Reply:
The clone is necessary, because we need one vector for the `MultiSpan` of the lint itself and one for the suggestion.

The collect is turning the `HashSet` into a `Vec` for the `MultiSpan`. But it would probably be better to just sort and dedup a vector instead of using a hash set, will change that.